pipelines/rbsa/tasks/zipcode_correlation.py

```diff
@@ -0,0 +1,67 @@
+import logging
+import pandas as pd
+from generics import task as t
+from generics.file_type_enum import SupportedFileReadType
+
+
+logger = logging.getLogger('LCTK_APPLICATION_LOGGER')
+
+
+class ZipcodeCorrelation(t.Task):
+    def __init__(self, name):
+        super().__init__(self)
+        self.name = name
+
+        self.input_artifact_full_zipcodes = 'full_zipcodes.csv'
+        self.input_artifact_projection_locations = 'PROJECTION_LOCATIONS.json'
+
+        # these will be used to generate list of input files
+        self.pre_data_files = [ 
+            { 'name': self.input_artifact_full_zipcodes, 'read_type': SupportedFileReadType.DATA },
+            { 'name': self.input_artifact_projection_locations, 'read_type': SupportedFileReadType.CONFIG }
+        ]
+
+        self.pre_data_map = self.load_data(self.pre_data_files) 
+        self.full_zipcodes = list(self.pre_data_map['full_zipcodes.csv']['zipcodes'])  
+        self.projection_locations = list(self.pre_data_map['PROJECTION_LOCATIONS.json']['cities'].keys())
+
+        self.data_files = []
+
+        for location in self.full_zipcodes:
+            self.data_files.append({ 'name': 'tmy_base/'+str(location)+'.csv', 'read_type': SupportedFileReadType.DATA })
+
+        for location in self.projection_locations:
```

Comment:
Not sure what the shape of these lists are but you could join the lists and loop once. Like:
```
joined_lists = self.full_zipcodes + self.project_locations
for location in joined_lists:
    ...
```

Reply:
fixed